File: 백준/Silver/3986. 좋은 단어/좋은 단어.cc
#include<bits/stdc++.h>
using namespace std;

int n,answer;
int main(){
    cin >> n;
    for(int i = 0;i<n;i++){
        string input;
        cin >> input;

        stack<char>stack;
        stack.push(input[0]);

        for(int i = 1;i<input.size();i++){
            if(stack.size() != 0){
                if(input[i] == stack.top()){
                    stack.pop();
                }else{
                    stack.push(input[i]);
                }
            }else{
                stack.push(input[i]);
            }
        }
        if(stack.size() == 0){
            answer ++;
        }
    }
    cout << answer << "\n";
}

File: 백준/Silver/3986. 좋은 단어/좋은 단어.java
import java.io.*;
import java.util.*;


public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        
        int answer = 0;

        for (int i = 0; i < n; i++) {
            Stack<Character> stack = new Stack<>();
            String input = br.readLine();

            for (char c : input.toCharArray()) {
                if (stack.isEmpty()) {
                    stack.push(c);
                    continue;
                }

                if (stack.peek() == c) {
                    stack.pop();
                } else {
                    stack.push(c);
                }
            }

            if (stack.isEmpty())
                answer++;

        }

        System.out.println(answer);
    }
}